openfl/component/collaborator/collaborator.py

```diff
@@ -171,6 +174,14 @@ def run(self):
 
                 # Cleaning tensor db
                 self.tensor_db.clean_up(self.db_store_rounds)
+                if self.log_memory_usage:
+                    # This is the place to check the memory usage of the collaborator
+                    memory_detail = get_memory_usage(
+                        self.logger, round_number, metric_origin=self.collaborator_name
+                    )
+                    memory_details.append(memory_detail)
+        if self.log_memory_usage:
+            self.logger.info(f"Publish memory usage: {self.memory_details}")
```

Comment:
I suggest logging once at the end of experiment, or only within the function call.